flow/core/rewards.py

```diff
@@ -330,3 +330,93 @@ def energy_consumption(env, gain=.001):
         power += M * speed * accel + M * g * Cr * speed + 0.5 * rho * A * Ca * speed ** 3
 
     return -gain * power
+
+
+def veh_energy_consumption(env, veh_id, gain=.001):
+    """Calculate power consumption of a vehicle.
+    Assumes vehicle is an average sized vehicle.
+    The power calculated here is the lower bound of the actual power consumed
+    by a vehicle.
+    """
+    power = 0
+
+    M = 1200  # mass of average sized vehicle (kg)
+    g = 9.81  # gravitational acceleration (m/s^2)
+    Cr = 0.005  # rolling resistance coefficient
+    Ca = 0.3  # aerodynamic drag coefficient
+    rho = 1.225  # air density (kg/m^3)
+    A = 2.6  # vehicle cross sectional area (m^2)
+    speed = env.k.vehicle.get_speed(veh_id)
+    prev_speed = env.k.vehicle.get_previous_speed(veh_id)
+
+    accel = abs(speed - prev_speed) / env.sim_step
+
+    power += M * speed * accel + M * g * Cr * speed + 0.5 * rho * A * Ca * speed ** 3
```

Comment:
do you have a reference especially for some of the coefficients (e.g., Cr)? if yes, you can add it to the docstring

Reply:
No, they're from Joy and are custom values